src/Symfony/Component/Messenger/Bridge/Kafka/Transport/KafkaReceiver.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Bridge\Kafka\Transport;
+
+use Symfony\Component\Messenger\Bridge\Kafka\Stamp\KafkaReceivedMessageStamp;
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\Exception\TransportException;
+use Symfony\Component\Messenger\Transport\Receiver\ReceiverInterface;
+use Symfony\Component\Messenger\Transport\Serialization\PhpSerializer;
+use Symfony\Component\Messenger\Transport\Serialization\SerializerInterface;
+
+class KafkaReceiver implements ReceiverInterface
+{
+    public function __construct(
+        private Connection $connection,
+        private SerializerInterface $serializer = new PhpSerializer(),
+    ) {
+    }
+
+    /** @psalm-return \Traversable<Envelope> */
+    public function get(): iterable
+    {
+        yield from $this->getEnvelope();
+    }
+
+    /** @SuppressWarnings(PHPMD.UnusedFormalParameter) */
+    public function ack(Envelope $envelope): void
+    {
+        /** @var KafkaReceivedMessageStamp $transportStamp */
+        $transportStamp = $envelope->last(KafkaReceivedMessageStamp::class);
+
+        $this->connection->ack($transportStamp->message);
+    }
+
+    /** @SuppressWarnings(PHPMD.UnusedFormalParameter) */
+    public function reject(Envelope $envelope): void
+    {
+        // no reject method for kafka transport
+    }
+
+    /** @psalm-return iterable<Envelope> */
+    private function getEnvelope(): iterable
```

Comment:
why making a private method that returns an iterable when we can only have 0 or 1 element ? Returning `Envelope|null` would make the code easier to understand.